src/components/projects/modules/saga.js

```diff
@@ -0,0 +1,68 @@
+import { takeLatest, put, call } from 'redux-saga/effects';
+import { fetchAllProjectsSuccess, fetchAllProjectsFailure, startLoader, stopLoader } from './actions';
+import { actionTypes } from "./types";
+import { getAllProjects } from '../../../service/projectsService';
+
+function generateQueryParams(selectedProjectStatus, searchInput) {
+  const statusQueryParam = selectedProjectStatus !== 'All Projects' ? selectedProjectStatus : '';
+  const searchQueryParam = searchInput ? `search[name]=${encodeURIComponent(searchInput)}` : '';
```

Comment:
Why you encode this. If user type name of project its visible already. He don't send anything private or important. he just type search